account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2017-2018 Vauxoo
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def action_account_change_currency(self):
+        track = self.env['mail.tracking.value']
+        today = fields.Date.today()
+        for invoice in self.filtered(lambda x: x.state == 'draft'):
+            from_currency = invoice.get_last_currency_id()
+            if not from_currency or from_currency == invoice.currency_id:
+                continue
+            ctx = {'company_id': invoice.company_id.id,
+                   'date': invoice.date_invoice or today}
+            currency = invoice.with_context(ctx).currency_id
```

Comment:
Use `**ctx` since that the method `with_context` create an empty context if the parameter is a dictionary, but update current context otherwise.

Reply:
Just for the record:

https://github.com/odoo/odoo/blob/45dec698ee2b113c19277dbc39abcb4f0897da16/odoo/models.py#L4698-L4719